### ISSUE TITLE
Update prql-lib, dotnet & php libraries

### ISSUE THREAD

**Issue by vanillajonathan** [PRQL/prql#1859]
The `prql-lib` library does not expose the [`compile`](https://docs.rs/prql-compiler/latest/prql_compiler/fn.compile.html) method.

**Comment by eitsupi** [comment]
I think this means that prql-lib has not actually been updated and the `compile` is still named `to_sql`, as it was before (0.3).
https://github.com/PRQL/prql/blob/main/CHANGELOG.md#040--2022-01-15

**Comment by vanillajonathan** [comment]
Can someone update the library and add the `compile` method?

I think that the `Options` struct in [prql-compiler/src/lib.rs](https://github.com/PRQL/prql/blob/main/prql-compiler/src/lib.rs) needs to use the `#[repr(C)]` macro and that [prql-lib/src/lib.rs](https://github.com/PRQL/prql/blob/main/prql-lib/src/lib.rs) needs to add something like:
```rust
#[no_mangle]
/// # Safety
///
/// This function is inherently unsafe because it is using C ABI.
pub unsafe extern "C" fn compile(query: *const c_char, options: &Options) -> *const c_char {
    let prql_query: String = CStr::from_ptr(query).to_string_lossy().into_owned();

    let result = match prql_compiler::compile(&prql_query, &options) {
        Ok(sql_str) => sql_str,
        Err(_) => "".to_string()
    };

    let c_string = CString::new(result).unwrap();
    
    c_string.into_raw()
}
```

But the `Options` struct also contain a `Target` which I don't know how to deal with.

Then it can be tested with this C program.
```c
#include <stdio.h>

#include <libprql_lib.h>

int main() {
    char *prql_query;
    prql_query = "from albums | select [album_id, title] | take 3";

    struct Options options;
    options.format = true;
    options.signature_comment = true;

    const char *sql_query = compile(prql_query, &options);

    printf("%s", sql_query);
    return 0;
}
```

The C header file `libprql_lib.h` needs to be updated but cbindgen complains.

    $ cargo install --force cbindgen
    $ cbindgen
    WARN: Can't find Options. This usually means that this type was incompatible or not found.

We would want `cbindgen` to add something like the following to the `libprql_lib.h` file.
```h
typedef struct Options {
    bool format;
    //Target target; // I don't know?
    bool signature_comment;
} Options;

/**
 * # Safety
 *
 * This function is inherently unsafe because it is using C ABI.
 */
const char *compile(const char *query, const Options *options);
```

Maybe @aljazerzen or @sigxcpu76 knows...

**Comment by eitsupi** [comment]
Target can be generated from string like that
https://github.com/PRQL/prql/blob/4d02cf35cd3e3a8100afece5d776a4cd9df25a0f/prql-python/src/lib.rs#L99

**Comment by aljazerzen** [comment]
I've updated the prql-lib, but now also PHP and dotnet libraries need to be updated.

I don't have that env set-up, can someone else do it?

**Comment by vanillajonathan** [comment]
I can do it!

**Comment by max-sixty** [comment]
Closed by #2111 & others. Thanks @vanillajonathan for helping!
